bokehjs/test/unit/regressions.ts

```diff
@@ -1239,4 +1240,81 @@ describe("Bug", () => {
       })
     })
   })
+
+  describe("in issue #13555", () => {
+    it("doesn't allow to computed correct image index for inverted ranges", async () => {
+      const n = 5
+
+      async function plot(x_flipped: boolean, y_flipped: boolean) {
+        const x = linspace(0, 10, n)
+        const y = linspace(0, 10, n)
+
+        const values: number[] = []
+        for (const yi of y) {
+          for (const xi of x) {
+            values.push(xi + yi)
+          }
+        }
+        const image = ndarray(values, {dtype: "float64", shape: [n, n]})
+
+        const x_range = new DataRange1d({flipped: x_flipped})
+        const y_range = new DataRange1d({flipped: y_flipped})
+
+        const p = fig([300, 300], {x_range, y_range, toolbar_location: "right"})
+
+        const color_mapper = new LinearColorMapper({palette: Viridis256})
+        const img = p.image({image: [image], x: 0, y: 0, dw: 2*n, dh: 2*n, color_mapper})
+
+        const hover = new TapTool({renderers: [img], behavior: "select"})
+        p.add_tools(hover)
+
+        const {view} = await display(p)
+        return {view, img}
+      }
+
+      function image_index(i: number, j: number) {
+        return [{index: 0, i, j, flat_index: j*n + i}]
+      }
+
+      async function test(options: {x_flipped: boolean, y_flipped: boolean}) {
+        const {x_flipped, y_flipped} = options
+
+        const {view, img} = await plot(x_flipped, y_flipped)
+        const actions = new PlotActions(view)
+
+        await actions.tap({x: 1, y: 1})
+        await view.ready
+        await delay(500) // remove in PR #12831
```

Comment:
I will wait with this PR till #12831 is merged, because otherwise I would either have to increase this number further or leave it, making the test unreliable.